### ISSUE TITLE
[G8] `trajectory_funnel_benchmark.py` returns hardcoded mock RL results — either implement or delete

### ISSUE THREAD

**Issue by dieterolson** [D-sorganization/Golf_Modeling_Suite#3168]
Part of #3160 (Adversarial Review Track G).

## Problem

`src/reinforcement_learning/trajectory_funnel_benchmark.py:63-77` contains a method named `simulate_agent_training_mock` whose body is:

```python
def simulate_agent_training_mock(self) -> dict[str, float]:
    """Mocks the RL convergence behavior discussed in Chapter 10.
    This will be replaced with Stable Baselines3 + MuJoCo in future PRs."""
    ...
    return {"convergence_epochs": 15000, "terminal_variance": 4.5}
```

Two problems:

1. **The values are hardcoded.** They do not come from a simulation. Any downstream report/dashboard that consumes them is lying to the reader.
2. **"This will be replaced"** — but that PR never landed. The file has sat in the repo as a placeholder that will be interpreted as real output by anyone who doesn't read line 63.

This is exactly the kind of scaffolding that #3062 targets, but is not called out by name there. G8 is the concrete file-level instance.

## Deliverable

Pick one:

### Option A: implement for real

Replace the mock with a real Stable Baselines3 + MuJoCo training loop:

1. Build or reuse a Gymnasium env for the funnel task. `src/learning/rl/base_env.py` already has Gymnasium scaffolding; extend it.
2. Train PPO or SAC for N epochs (configurable). Record convergence epoch (first epoch where reward is within ε of max).
3. Return real metrics with provenance: `{convergence_epochs: int, terminal_variance: float, seed: int, engine: str, commit_sha: str, training_duration_s: float}`.
4. Cache results on disk keyed by `(env_version, algo_version, seed)` so re-runs are cheap.
5. Gate the heavy path under `@pytest.mark.slow` so CI doesn't run it by default.
6. Provide a reproducible Jupyter / CLI entrypoint — e.g., `python -m src.reinforcement_learning.trajectory_funnel_benchmark --algo ppo --epochs 1000 --seed 0`.

### Option B: delete and land a placeholder issue

If real training is out of scope:

1. Delete `trajectory_funnel_benchmark.py`.
2. Audit `src/reinforcement_learning/` and `src/learning/` for all references to the mock (`grep -rn "simulate_agent_training_mock\\|trajectory_funnel"`).
3. Remove any dashboards/reports that consume the mock output.
4. File a new issue "Implement trajectory funnel benchmark" documenting requirements and mark this issue closed as `wontfix-for-now`.

**Do not** leave the mock in place. `success: True` + placeholder data is exactly the pattern that erodes trust in the whole suite.

## Acceptance criteria

- [ ] `grep -rn "simulate_agent_training_mock" src/` returns zero hits, **or** the method's body runs a real training loop and the hardcoded dict is gone.
- [ ] No tests pass by asserting the hardcoded `{convergence_epochs: 15000}` value.
- [ ] Any consumer of the output that assumed real numbers (grep for callers of the benchmark class) has been updated.

## Similar audit

While you're here, grep the rest of `src/` for analogous patterns and call them out in the PR description:

```bash
grep -rEn "mock|hardcoded|placeholder|simulation_pending|will be replaced" src/
```

Patterns to flag:
- Functions named `*_mock` that are imported from non-test code.
- Docstrings containing "This will be replaced".
- Dicts with round-number hardcoded values returned from "benchmark" or "training" functions.

## References

- Mock: `src/reinforcement_learning/trajectory_funnel_benchmark.py:63-77`
- RL scaffolding (real): `src/learning/rl/base_env.py`
- Related umbrella thread: #3062 (prove robotics / learning / research subsystems are real)
